src/main/java/org/opensearch/ad/EntityProfileRunner.java

```diff
@@ -132,10 +135,56 @@ public void profile(
         }, listener::onFailure));
     }
 
+    private void validateEntity(
+        Entity entity,
+        List<String> categoryFields,
+        String detectorId,
+        Set<EntityProfileName> profilesToCollect,
+        AnomalyDetector detector,
+        ActionListener<EntityProfile> listener
+    ) {
+        Map<String, String> attributes = entity.getAttributes();
+        if (attributes == null || attributes.size() != categoryFields.size()) {
+            listener.onFailure(new InvalidParameterException("Empty entity attributes"));
+            return;
+        }
+        for (String field : categoryFields) {
+            if (false == attributes.containsKey(field)) {
+                listener.onFailure(new InvalidParameterException("Cannot find " + field));
+                return;
+            }
+        }
+
+        BoolQueryBuilder internalFilterQuery = QueryBuilders.boolQuery().filter(detector.getFilterQuery());
+
+        for (TermQueryBuilder term : entity.getTermQueryBuilders()) {
+            internalFilterQuery.filter(term);
+        }
+
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().query(internalFilterQuery).size(1);
+
+        SearchRequest searchRequest = new SearchRequest(detector.getIndices().toArray(new String[0]), searchSourceBuilder)
+            .preference(Preference.LOCAL.toString());
+
+        client.search(searchRequest, ActionListener.wrap(searchResponse -> {
```

Comment:
Need to do it in case of typos for example.  The entity is given by the API user.  Not sure it actually exists or not. Added a method comment to explain.

Reply:
If user delete the entity after job start, then we will not be able to get this entity in index. For this case, we will not return profile for this entity even it's running on some data node?

Reply:
yes, the entity's model will be deleted by other entity or by maintenance due to long inactivity.